### PR TITLE
create interfaces for workers to make testing more direct

### PR DESCRIPTION
**Pull request by brettfo**
This makes testing `RunWorker.cs` much faster; approx 20s to 1s.

It also fixes some path handling on Windows machines; common for development, but not production or CI.

### CODE REVIEW

File: nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs [hunk 1]
```diff
@@ -202,17 +216,16 @@ await _apiHandler.IncrementMetric(new()
                                     Groups = previousDependency.Requirements.Single().Groups,
                                     Source = new RequirementSource()
                                     {
-                                        SourceUrl = analysisResult.UpdatedDependencies.Single(d => d.Name == dependency.Name).InfoUrl,
+                                        SourceUrl = analysisResult.UpdatedDependencies.FirstOrDefault(d => d.Name == dependency.Name)?.InfoUrl,
```
Comment (sebasgomez238) [inline nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs:219]: Just for my own learning, why is this needed?

Comment (brettfo) [reply nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs:219]: In case it's not found, this will allow `null` to flow through instead of throwing.  It _should_ always be present, but if it's not and we can prevent the throw then we can at least continue with the update and PR creation.
